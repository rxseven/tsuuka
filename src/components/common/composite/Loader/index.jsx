import React from 'react';
import styled from 'styled-components';

import { Body, Document, Head } from 'components/common/base/Page';
import Button from 'components/common/base/Button';
import Card from 'components/common/base/Card';
import Spinner from 'components/common/base/Spinner';
import Layout from 'components/common/composite/Layout';

const Indicator = styled.div`
  text-align: center;
`;

const Message = styled.div`
  color: ${({ theme }) => theme.color.secondary};
  font-size: 0.75rem;
  padding-top: 1rem;
  text-align: center;
`;

function Loader(props) {
  const { error, pastDelay, timedOut } = props;
  let content = null;

  switch (true) {
    // When the loader has errored
    case error:
      content = (
        <Card>
          <Card.Header>Sorry</Card.Header>
          <Card.Body>
            <Card.Text>
              Something went wrong, please reload a webpage.
            </Card.Text>
            <Button
              id="loader-reload"
              look="primary"
              onClick={() => window.location.reload()}
            >
              Reload
            </Button>
          </Card.Body>
        </Card>
      );
      break;

    // When the loader has taken longer than the timeout
    case timedOut:
      content = (
        <Indicator>
          <Spinner />
          <Message>
            <p>Please take a moment</p>
          </Message>
        </Indicator>
      );
      break;

    // When the loader has taken longer than the delay
    case pastDelay:
      content = (
        <Indicator>
          <Spinner />
        </Indicator>
      );
      break;

    // When the loader has just started
    default:
      content = null;
  }

  // Return content if any
  if (content) {
    return (
      <Document>
        <Head>
          <title>通貨換算ツール</title>
        </Head>
        <Body>
          <Layout adaptive>{content}</Layout>
        </Body>
      </Document>
    );
  }

  // Return nothing
  return content;
}

export default Loader;
