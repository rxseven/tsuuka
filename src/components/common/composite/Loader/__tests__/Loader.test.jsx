import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../index';

describe('<Loader />', () => {
  // Arrange
  const component = <Loader />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
