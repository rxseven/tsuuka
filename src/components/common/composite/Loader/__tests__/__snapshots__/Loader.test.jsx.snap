// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Snapshot tests should render correctly 1`] = `null`;
